### PR TITLE
Profiles API

### PR DESCRIPTION
**Pull request by abir499-ban**
This PR fixes issue #13  and configure he Profiles API along with a well structured mongoose Schema keeping in mind the future expansions.

### CODE REVIEW

**Comment by majorbruteforce** [comment]
Update the dto according to schema

File: src/services/apis/profiles/schemas/profiles.schema.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+export type ProfilesDocument = HydratedDocument<Profiles>;
+
+@Schema({
+  timestamps: true,
+})
+export class Profiles extends SoftDeleteSchema {
+  @Prop({
+    type: String,
+    enum: ['Football', 'Cricket', 'Badminton', 'Basketball', 'Volleyball'],
```
Comment (zakhaev26) [inline src/services/apis/profiles/schemas/profiles.schema.ts]: suggestion: this should be an enum type , present in a common folder.

File: src/services/apis/profiles/schemas/profiles.schema.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    ref: Users.name,
+    index: true,
+    required: true,
+    set: EnsureObjectId,
+  })
+  user: Types.ObjectId;
+
+  @Prop({
+    type: String,
+    required: true,
```
Comment (zakhaev26) [inline src/services/apis/profiles/schemas/profiles.schema.ts]: remove `required:true`

not mandatory for every sport

File: src/services/apis/profiles/dto/profiles.dto.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      description: z.string().max(150).min(25).optional(),
+    })
+  ).optional(),
+
+  createdBy: z
+    .string()
+    .refine((val) => Types.ObjectId.isValid(val), {
+      message: "invalid user ID"
+    })
+});
```
Comment (zakhaev26) [inline src/services/apis/profiles/dto/profiles.dto.ts]: each field should be `optional()` during patch , and remove defaults
learn more: https://zod.dev/?id=optional

**Review by zakhaev26 — CHANGES_REQUESTED** [review]

**Review by zakhaev26 — APPROVED** [review]
LGTM
Thanks for the feature! 🎉

**Pull request merged**